src/status.cc

```diff
@@ -14,172 +14,133 @@
-    case Status::JwksNotImplementedKty:
-      return "[kty] is not supported in [keys]";
-
-    case Status::JwksRSAKeyBadAlg:
-      return "[alg] is not started with [RS] or [PS] for an RSA key";
-    case Status::JwksRSAKeyMissingN:
-      return "[n] field is missing for a RSA key";
-    case Status::JwksRSAKeyBadN:
-      return "[n] field is not string for a RSA key";
-    case Status::JwksRSAKeyMissingE:
-      return "[e] field is missing for a RSA key";
-    case Status::JwksRSAKeyBadE:
-      return "[e] field is not string for a RSA key";
-
-    case Status::JwksECKeyBadAlg:
-      return "[alg] is not started with [ES] for an EC key";
-    case Status::JwksECKeyBadCrv:
-      return "[crv] field is not string for an EC key";
-    case Status::JwksECKeyAlgOrCrvUnsupported:
-      return "[crv] or [alg] field is not supported for an EC key";
-    case Status::JwksECKeyAlgNotCompatibleWithCrv:
-      return "[crv] field specified is not compatible with [alg] for an EC key";
-    case Status::JwksECKeyMissingX:
-      return "[x] field is missing for an EC key";
-    case Status::JwksECKeyBadX:
-      return "[x] field is not string for an EC key";
-    case Status::JwksECKeyMissingY:
-      return "[y] field is missing for an EC key";
-    case Status::JwksECKeyBadY:
-      return "[y] field is not string for an EC key";
-
-    case Status::JwksHMACKeyBadAlg:
-      return "[alg] does not start with [HS] for an HMAC key";
-    case Status::JwksHMACKeyMissingK:
-      return "[k] field is missing for an HMAC key";
-    case Status::JwksHMACKeyBadK:
-      return "[k] field is not string for an HMAC key";
-
-    case Status::JwksOKPKeyBadAlg:
-      return "[alg] is not [EdDSA] for an OKP key";
-    case Status::JwksOKPKeyMissingCrv:
-      return "[crv] field is missing for an OKP key";
-    case Status::JwksOKPKeyBadCrv:
-      return "[crv] field is not string for an OKP key";
-    case Status::JwksOKPKeyCrvUnsupported:
-      return "[crv] field is not supported for an OKP key";
-    case Status::JwksOKPKeyMissingX:
-      return "[x] field is missing for an OKP key";
-    case Status::JwksOKPKeyBadX:
-      return "[x] field is not string for an OKP key";
-
-    case Status::JwksX509BioWriteError:
-      return "X509 parse pubkey internal fails: memory allocation";
-    case Status::JwksX509ParseError:
-      return "X509 parse pubkey fails";
-    case Status::JwksX509GetPubkeyError:
-      return "X509 parse pubkey internal fails: get pubkey";
-
-    case Status::JwksPemNotImplementedKty:
-      return "PEM Key type is not supported";
-    case Status::JwksPemBadBase64:
-      return "PEM pubkey parse fails";
-    case Status::JwksPemGetRawEd25519Error:
-      return "PEM failed to get raw ED25519 key";
-
-    case Status::JwksBioAllocError:
-      return "Failed to create BIO due to memory allocation failure";
-  };
-  return "";
+  static const auto* status_str_map = new absl::flat_hash_map<
```

Comment:
Why change from switch-case to map? Don't we want exhaustive list? IIRC switch-case in C++ requires all values to be defined, there is no chance of skipping one. 

Do note [go/totw/147](https://goto.google.com/totw/147) but I feel it's OK since we own this library

Reply:
Is it to increase code coverage? TBH I feel it is not the right approach. Feels like we are trying to hack around the code coverage system. 

I prefer not having any coverage. Anyways we have the following guidance in https://cloud.google.com/apis/design/errors#error_messages

> Warning: Error messages are not part of the API surface. They are subject to changes without notice. Application code must not have a hard dependency on error messages.

We can argue there is no need to explicitly test for them